### PR TITLE
Functional tests - Add 'data-role' attribute to 'header-desc' selectors

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add 'data-role' attribute to 'header-desc' selectors
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

File: admin-dev/themes/default/template/page_header_toolbar.tpl [hunk 1]
```diff
@@ -101,7 +102,8 @@
                   id="page-header-desc-{$table}-{if isset($toolbar_btn['modules-list'].imgclass)}{$toolbar_btn['modules-list'].imgclass}{else}modules-list{/if}"
                   class="toolbar_btn{if isset($toolbar_btn['modules-list'].class)} {$toolbar_btn['modules-list'].class}{/if}{if isset($toolbar_btn['modules-list'].target) && $toolbar_btn['modules-list'].target} _blank{/if}"
                   {if isset($toolbar_btn['modules-list'].href)}href="{$toolbar_btn['modules-list'].href}"{/if}
-                  title="{$toolbar_btn['modules-list'].desc}"{if isset($toolbar_btn['modules-list'].js) && $toolbar_btn['modules-list'].js} onclick="{$toolbar_btn['modules-list'].js}"{/if}>
+                  title="{$toolbar_btn['modules-list'].desc}"{if isset($toolbar_btn['modules-list'].js) && $toolbar_btn['modules-list'].js} onclick="{$toolbar_btn['modules-list'].js}"{/if}
+                  data-role="page-header-desc-{$table}">
```
Comment (boubkerbribri) [inline admin-dev/themes/default/template/page_header_toolbar.tpl]:
```suggestion
                  data-role="page-header-new-{$table}-link">
```

Better this way i think

File: tests/UI/pages/BO/catalog/attributes/index.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Attributes extends BOBasePage {
 
     // Header selectors
     this.addNewAttributeLink = '#page-header-desc-attribute_group-new_attribute_group';
-    this.addNewValueLink = '#page-header-desc-attribute_group-new_value';
+    this.addNewValueLink = '// a[data-role="page-header-desc-attribute_group"]';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/attributes/index.js]:
```suggestion
    this.addNewValueLink = 'a[data-role="page-header-desc-attribute_group"]';
```

I think it was xpath that you used

File: tests/UI/pages/BO/catalog/discounts/catalogPriceRules/index.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ class CatalogPriceRules extends BOBasePage {
     this.pageTitle = 'Catalog Price Rules •';
 
     // Selectors header
-    this.addNewCatalogPriceRuleButton = '#page-header-desc-specific_price_rule-new_specific_price_rule';
+    this.addNewCatalogPriceRuleButton = 'a[data-role="page-header-desc-specific_price_rule"]';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/discounts/catalogPriceRules/index.js]:
```suggestion
    this.addNewCatalogPriceRuleButton = 'a[data-role=page-header-desc-specific_price_rule]';
```

You don't need the double quote, you can delete it everywhere

File: tests/UI/pages/BO/catalog/attributes/index.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/UI/pages/BO/catalog/attributes/index.js]:
```suggestion
    this.addNewValueLink = '[data-role=page-header-desc-attribute_group]';
```

I think you can even delete the tag

File: tests/UI/pages/BO/catalog/discounts/catalogPriceRules/index.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/discounts/catalogPriceRules/index.js]: Donc thx :+1:

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**